### PR TITLE
code refractoring and unit testing

### PR DESCRIPTION
**Pull request by SrinivasanTarget**
Raising this instead of #65

### CODE REVIEW

**Comment by SrinivasanTarget** [comment]
@imurchie FYI

File: lib/commands/general.js [hunk 1]
```diff
@@ -67,7 +68,22 @@ commands.getPageSource = async function () {
 };
 
 commands.background = async function (secs) {
-  await this.uiAutoClient.sendCommand(`au.background(${secs})`);
```
Comment (imurchie) [inline lib/commands/general.js]: This is overwriting a change that is needed.

**Comment by SrinivasanTarget** [comment]
Updated!

**Comment by imurchie** [comment]
That function ought to look like https://github.com/appium/appium-ios-driver/blob/master/lib/commands/general.js#L69-L71

**Comment by SrinivasanTarget** [comment]
Yes updated it now

**Comment by SrinivasanTarget** [comment]
Squashed now

**Comment by imurchie** [comment]
Looks good. We'll let CI run and then get it in! Sorry for the slowness of the process.

**Comment by imurchie** [comment]
Awesome! Merging! Thank you for bearing with us through this.

**Comment by SrinivasanTarget** [comment]
Thanks @imurchie What's the next step on this?Shall i proceed with java client implementation?

**Comment by imurchie** [comment]
Yes. I will make an issue in each of the clients, to create a method to use this endpoint. You can begin the Java implementation if that is your language of choice.

**Comment by SrinivasanTarget** [comment]
(y)

**Comment by imurchie** [comment]
I've created an issue in the `mobile-json-wire-protocol` repo to track this progress.

https://github.com/appium/node-mobile-json-wire-protocol/issues/40

**Comment by SrinivasanTarget** [comment]
Thanks @imurchie

**Pull request merged**